compiler/passes/filesToAST.cpp

```diff
@@ -220,3 +239,49 @@ static void gatherWellKnownTypes() {
     }
   }
 }
+
+static void gatherWellKnownFns() {
+  int nEntries = sizeof(sWellKnownFns) / sizeof(sWellKnownFns[0]);
+  static const char* mult_def_message   = "'%s' defined more than once in Chapel internal modules.";
+  static const char* flag_reqd_message = "The '%s' function is missing a required flag.";
+  static const char* wkfn_reqd_message   = "Function '%s' must be defined in the Chapel internal modules.";
+
+  // Harvest well-known functions from among the global fn definitions.
+  // We check before assigning to the well-known type dt<typename>, to ensure that it
+  // is null.  In that way we can flag duplicate definitions.
+  forv_Vec(FnSymbol, fn, gFnSymbols) {
+    for (int i = 0; i < nEntries; ++i) {
+      WellKnownFn& wkfn = sWellKnownFns[i];
+
```

Comment:
This seems to imply that it's possible for there to be multiple uses of a name where precisely 1 of them has the expected flag.  Is that a real case?  If not then this seems unnecessarily convoluted.  Otherwise OK.